app/jobs/huutokauppa_job.rb

```diff
@@ -2,11 +2,59 @@ class HuutokauppaJob < ActiveJob::Base
   queue_as :huutokauppa
 
   def perform(id:)
-    incoming_mail = IncomingMail.find(id)
+    @incoming_mail = IncomingMail.find(id)
 
-    incoming_mail.update(
-      processed_at: Time.now,
-      status: :ok
-    )
+    Current.company = @incoming_mail.company
+    Current.user    = Current.company.users.find_by!(kuka: 'admin')
+
+    @huutokauppa_mail = HuutokauppaMail.new(@incoming_mail.raw_source)
+
+    case @huutokauppa_mail.type
+    when :offer_accepted, :offer_automatically_accepted
+      update_order_customer_and_product_info
+    when :purchase_price_paid
+      mark_order_as_done_and_set_delivery_method
+    when :delivery_ordered
+      update_order_delivery_info_and_delivery_method
+    end
   end
+
+  private
+
+    def update_order_customer_and_product_info
+      customer = @huutokauppa_mail.find_customer || @huutokauppa_mail.create_customer
```

Comment:
Tässä on turha muuttuja `customer`. Lisäksi voisi olla selkeämpää, että tämäkin olisi `HuutokauppaMail`:in asioita, ts. tämä olisi vain `@huutokauppa_mail.update_customer`, joka sitten hoitaisi tuon find || create logiikan.